canopy_config.ml

```diff
@@ -21,14 +13,28 @@ let decompose_git_url url =
 let remote_uri () = fst (decompose_git_url (Key_gen.remote ()))
 let remote_branch () = snd (decompose_git_url (Key_gen.remote ()))
 
-let config () = {
-  remote_uri = remote_uri ();
-  remote_branch = remote_branch ();
-  index_page = Key_gen.index ();
-  blog_name = Key_gen.name ();
-  port = Key_gen.port ();
-  push_hook_path = Key_gen.push_hook ();
-  tls_port = Key_gen.tls_port ();
-  uuid = Key_gen.uuid ();
-  root = Key_gen.root ();
-}
+let index_page cache =
+  match KeyMap.find_config_opt cache [".config";"index_page"] with
+    None -> "Index"
+  | Some p -> p
+
+let blog_name cache =
+  match KeyMap.find_config_opt cache [".config";"blog_name"] with
+    None -> "Canopy"
+  | Some n -> n
+
+let uuid cache =
+  match KeyMap.find_config_opt cache [".config";"uuid"] with
+    None -> raise (Required_config "uuid")
+  | Some u -> u
```

Comment:
At the moment this function is only called from within `Canopy_syndic.atom` this would occur whenever `update_atom` is used... so it is not limited to startup time (for instance it happens if we request the `push_hook_path`)